syntax = "proto3";

package hconsole;

option java_multiple_files = true;
option java_package = "io.h001.hconsole.grpc";
option java_outer_classname = "StreamProto";

message NodeInfo {
  string id = 1;
  string ip = 2;
  string hostname = 3;
  string role = 4;
  double cpuUsage = 5;
  double memoryUsage = 6;
  string status = 7;
  string uptime = 8;
}

message Metric {
  repeated NodeInfo nodes = 1;
  double timestamp = 2;
}

message Log {
  double timestamp = 1;
  string level = 2;
  string message = 3;
}

message StreamPayload {
  oneof payload {
    Metric metric = 1;
    Log log = 2;
  }
}

message Response {
  bool success = 1;
}

service StreamService {
  rpc SendStream(stream StreamPayload) returns (Response);
}